### ISSUE TITLE
Reproducing the results of the paper

### ISSUE THREAD

**Issue by JanS97** [RyanNavillus/reward-surfaces#3]
Hi,

I tried reproducing the results of the paper on HalfCheetah-v2. I ran the following commands.

```
python3 scripts/train_agent.py "./runs/halfcheetah_checkpoints" SB3_ON HalfCheetah-v2 cuda '{"ALGO": "PPO"}' --save_freq=10000
python3 scripts/generate_plane_jobs.py --grid-size=31 --magnitude=1.0 --num-steps=200000 "runs/halfcheetah_checkpoints/0200000" "runs/halfcheetah_surface"
python3 scripts/run_jobs_multiproc.py --num-cpus=14 "runs/halfcheetah_surface/jobs.sh"
python3 scripts/job_results_to_csv.py "runs/halfcheetah_surface"
python3 scripts/plot_plane.py "runs/halfcheetah_surface/results.csv" --outname="runs/halfcheetah" --env_name="HalfCheetah-v2"
```
These are the commands from the README except that I loaded the checkpoint at 200000 steps, which was also done in the paper according to table 2. I ran the experiment two times and found that the results of both runs look quite different from the results reported in figure 11.

![halfcheetah_episoderewards_3dsurface_1](https://user-images.githubusercontent.com/33448112/182103385-523bd1b1-f3f3-43e7-b9ac-88d0b5772fea.png)
![halfcheetah_episoderewards_3dsurface_2](https://user-images.githubusercontent.com/33448112/182103386-5c0302ab-60fe-467e-8ad7-f9493493f172.png)

So I was wondering whether there are any hyperparameters that I missed or whether I need to do anything else different to obtain results similar to those in the paper.

**Comment by RyanNavillus** [comment]
Hi, I believe our results look different because we used the trained checkpoint after 1 million timesteps, not 200,000. I assume the agents you show here have not converged to the same policy that we found in our paper yet. Table 2 in our paper refers to the number of evaluation steps used to evaluate each point in the plot. The correct number of training steps for each environment can be found in the hyperparameters folder. If you don't change any settings, it will use the correct number (according to RL Zoo) by default.

**Comment by JanS97** [comment]
Ok, then I misinterpreted Table 2. Thanks for clarifying.
Just to be sure: If I don't change any settings, the algorithm chooses the best performing checkpoint (runs/halfcheetah_checkpoints/best). Is this how the plots in the paper were generated?

**Comment by RyanNavillus** [comment]
Yes that is correct. We plotted the best checkpoint for each environment after training with the hyperparameters found in the hyperparameters folder. The settings in Table 2 are for the `generate_plane_jobs.py` script

**Comment by JanS97** [comment]
I ran the code for `runs/halfcheetah_checkpoints/best` but the results still look different from those of the paper. Below are the results of two runs in this configuration.
![halfcheetah_best_00_episoderewards_3dsurface](https://user-images.githubusercontent.com/33448112/185042944-71204469-8792-4776-910d-f52f2c759bcf.png)
![halfcheetah_best_01_episoderewards_3dsurface](https://user-images.githubusercontent.com/33448112/185042951-dd2ed577-822e-4019-8911-86a41cd8a9ae.png)

**Comment by RyanNavillus** [comment]
I believe those are almost correct, I just chose to plot these in a linear scale instead of the logarithmic scale (which is the default for reward scales this large) in the paper. You can disable the logarithmic scale by passing the flag `--logscale off` to the `plot_plane.py` script.

**Comment by JanS97** [comment]
Ah, I missed that. With the `--logscale off` argument the plots look similar to those in the paper.

![halfcheetah_best_00_linear_episoderewards_3dsurface](https://user-images.githubusercontent.com/33448112/185306161-d4b8ade1-b4c9-4d3b-9146-c81c5283f335.png)
![halfcheetah_best_01_linear_episoderewards_3dsurface](https://user-images.githubusercontent.com/33448112/185306180-1d2273bb-7f97-4ea4-b96c-2ceebc2ea847.png)

Thanks for your help!
